### PR TITLE
[Fix] PARSeq tensorflow fixes

### PR DESCRIPTION
**Pull request by felixdittrich92**
This PR:

- several fixes

- [x] test run

torch fail will be fixed if #1227 is merged

Validation set loaded in 0.002758s (2520 samples in 79 batches)
Train set loaded in 0.003837s (126000 samples in 3937 batches)
Validation loss decreased inf --> 4.64792: saving state...                                                                                                      
Epoch 1/20 - Validation loss: 4.64792 (Exact: 0.08% | Partial: 0.08%)
Validation loss decreased 4.64792 --> 4.37974: saving state...                                                                                                  
Epoch 2/20 - Validation loss: 4.37974 (Exact: 0.48% | Partial: 0.91%)
Validation loss decreased 4.37974 --> 4.3734: saving state...                                                                                                   
Epoch 3/20 - Validation loss: 4.3734 (Exact: 0.99% | Partial: 1.39%)
Validation loss decreased 4.3734 --> 4.32947: saving state...                                                                                                   
Epoch 4/20 - Validation loss: 4.32947 (Exact: 2.14% | Partial: 3.10%)

### CODE REVIEW

File: doctr/models/recognition/parseq/tensorflow.py [hunk 1]
```diff
@@ -362,36 +326,69 @@ def call(
...
+                    # remove the [EOS] tokens for the succeeding perms
+                    if i == 1:
+                        gt_out = tf.where(tf.equal(gt_out, self.vocab_size), self.vocab_size + 2, gt_out)
+                        n = tf.reduce_sum(tf.cast(tf.math.not_equal(gt_out, self.vocab_size + 2), dtype=tf.float32))
+
+                loss /= loss_numel
+
             else:
-                # eval step - use non-autoregressive decoding while training evaluation
-                logits = self.decode_non_autoregressive(features)
```
Comment (charlesmindee) [inline doctr/models/recognition/parseq/tensorflow.py:392]: why do you remove the decode_non_autoregressive in that case ?

**Review by charlesmindee — COMMENTED** [review]
Thanks, just one point to clarify

**Review by charlesmindee — DISMISSED** [review]
thanks, LGTM

**Review by charlesmindee — APPROVED** [review]
